crates/router/src/utils/user/sample_data.rs

```diff
@@ -52,18 +52,19 @@ pub async fn generate_sample_data(
 
     let business_label_default = merchant_parsed_details.first().map(|x| x.business.clone());
 
-    let profile_id = crate::core::utils::get_profile_id_from_business_details(
-        business_country_default,
-        business_label_default.as_ref(),
-        &merchant_from_db,
-        req.profile_id.as_ref(),
-        &*state.store,
-        false,
-    )
-    .await
-    .change_context(SampleDataError::InternalServerError)
-    .attach_printable("Failed to get business profile")?;
+    let business_profile = match state
+        .store
+        .list_business_profile_by_merchant_id(&merchant_id)
+        .await
+        .change_context(SampleDataError::InternalServerError)
+        .attach_printable("Failed to get business profile")
+        .map(|profiles| profiles.first().cloned())?
+    {
+        Some(profile) => profile,
+        None => return Err(SampleDataError::InternalServerError.into()),
+    };
```

Comment:
let the previous logic be there, add this logic of listing business profiles on top of that

Reply:
This is because, let's have the option of passing the profile_id in the request, we might need it later on.